### ISSUE TITLE
how to compile and run this project?

### ISSUE THREAD

**Issue by xiaobing395845388** [ONLYOFFICE/DocumentServer#641]
**Do you want to request a *feature* or report a *bug*?**

**What is the current behavior?**

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

**What is the expected behavior?**

**Did this work in previous versions of DocumentServer?**

**DocumentServer version:**

**Operating System:**

**Browser version:**

**Comment by ShockwaveNN** [comment]
https://helpcenter.onlyoffice.com/server/linux/document/compile-source-code.aspx

And working Docker examples based on this instruction
https://github.com/onlyoffice-testing-robot/documentserver-from-source-build

**Comment by SebAlbert** [comment]
I have several issues with this.

First, I am wondering why all those GNOME / GTK libraries, and even something related to the X Screensaver, are necessary. This seems really strange.

Second, why are there sitll 3rd party libraries and tools being downloaded (partially as binaries) and built instead of listing them as dependencies? Especially v8 is something I just installed previously (i.e., nodejs) according to the instructions.

Third, the "make.sh" fails on my system already in the first subfolder, "boost". Its spitting errors such as `error: "none" is not a known value of feature <optimization>` and not finding `stage/lib/*`.

**Comment by ShockwaveNN** [comment]
> First, I am wondering why all those GNOME / GTK libraries, and even something related to the X Screensaver, are necessary. This seems really strange.

Build stages are shared for DocumentServer and DesktopEditors, so it's just list of all dependencies, if you wanna - you can clean up this list and we update it.

> Second, why are there sitll 3rd party libraries and tools being downloaded (partially as binaries) and built instead of listing them as dependencies? Especially v8 is something I just installed previously (i.e., nodejs) according to the instructions.

We have a wide range of variants of Linux we are working on, and this is better solution to guarantee have exact version of 3dparty lib on any system

> Third, the "make.sh" fails on my system already in the first subfolder, "boost". Its spitting errors such as error: "none" is not a known value of feature <optimization> and not finding stage/lib/*

Please add info about your OS, and I try to investiage it.

**Comment by SebAlbert** [comment]
Thanks for the quick response!

I'm using an up-to-date installation of Gentoo Linux based on GCC version 8.3.0.
It seems 1_58 is quite an old version of boost.

**Comment by ShockwaveNN** [comment]
> It seems 1_58 is quite an old version of boost.

Yes, it's some of our base dependencies and updating this may break some old os, like WinXP. Not sure it's the case, but may be something like this. We understand this is a problem, but currently have no perfect solution

> GCC version 8.3.0

I tryed to compile DocumentServer on Gentoo, but out of lack, compiling of v8 on newer version of gcc is not an easy task
See https://github.com/onlyoffice-testing-robot/documentserver-from-source-build/pull/9

**Comment by SebAlbert** [comment]
> Yes, it's some of our base dependencies and updating this may break some old os, like WinXP. Not sure it's the case, but may be something like this. We understand this is a problem, but currently have no perfect solution

Are there known issues with Windows XP and newer boost or V8 versions, or are you just afraid of the possibility? Are there end-to-end tests or similar things in place so that it could be tried? It's probably a bad idea to stay in he past with some old versions forever.

**Comment by ShockwaveNN** [comment]
> Are there known issues with Windows XP and newer boost or V8 versions, or are you just afraid of the possibility?

I cannot recall correctly, but I try to raise question with our dev team about updating some of our core components, or at least find a reason why they fixed on such versions

**Comment by ShockwaveNN** [comment]
@SebAlbert 

> Third, the "make.sh" fails on my system already in the first subfolder, "boost".

I just actualize my [gentoo](https://github.com/onlyoffice-testing-robot/documentserver-from-source-build/pull/9) image, I'm not checking full compilation (I think it'll fail on v8 as before) now, but `boost` compiling is OK
